functions/chatbot/helperFunctions.js

```diff
@@ -0,0 +1,108 @@
+const twilio = require("twilio"); // needed in verifyTwilio() for verification
+
+function verifyTwilio(req) {
+  // as per https://www.twilio.com/blog/how-to-secure-twilio-webhook-urls-in-nodejs
+  const twilioSignature = req.headers["x-twilio-signature"];
+  const params = req.body;
+  const baseUrl =
+    "https://us-central1-hospitalcommunity.cloudfunctions.net/twilio";
+  const url = baseUrl + req.path;
+  const requestIsValid = twilio.validateRequest(
+    process.env.TWILIO_AUTH_TOKEN,
+    twilioSignature,
+    url,
+    params
+  );
+  return requestIsValid;
+}
+
+function searchByZip(hospital_index, zip) {
+  let results = [];
+  let city_hits = hospital_index.zip_index[zip] || [];
+  for (var i = 0; i < city_hits.length; i++) {
+    let h = hospital_index.id_index[city_hits[i]];
+    let id = city_hits[i];
+    results.push({ h, id });
+  }
+  return results;
+}
+
+// creates a new response object
+function tResponseNew() {
```

Comment:
what does the t mean here?

Reply:
So Twilio expects a response back in a specific JSON format. There are several 'action blocks' you can add to the response format to get it to various things. I created these function to make brainless to create/add these blocks.

So tResponseNew() creates a new object that has an "actions" array. And then subsequent t----() functions then add more blocks into that actions array. Trying to make it so that the Twilio syntax (which is pretty hard to debug, because Twilio just gives us a "invalid response JSON" log) is less of an issue when building out more complex chat responses.